graphql-dgs-client/src/main/kotlin/com/netflix/graphql/dgs/client/codegen/GraphQLQueryRequest.kt

```diff
@@ -63,10 +57,41 @@ class GraphQLQueryRequest(private val query: GraphQLQuery, private val projectio
         if (projection is BaseSubProjectionNode<*, *>) {
             builder.append(projection.root().toString())
         } else if (projection != null) {
+
             builder.append(projection.toString())
         }
 
         builder.append(" }")
         return builder.toString()
     }
 }
+
+class InputValueSerializer(private val scalars: Map<Class<*>, Coercing<*, *>> = emptyMap()) {
+    fun serialize(input: Any?): String? {
+        if (input == null) {
+            return null
+        }
+
+        val type = input::class.java
+
+        return if (scalars.contains(type)) {
+            """"${scalars[type]!!.serialize(input)}""""
+        } else if (type.isPrimitive || type.isAssignableFrom(java.lang.Integer::class.java) || type.isAssignableFrom(java.lang.Long::class.java) || type.isAssignableFrom(java.lang.Double::class.java) || type.isAssignableFrom(java.lang.Float::class.java) || type.isAssignableFrom(java.lang.Boolean::class.java) || type.isAssignableFrom(java.lang.Short::class.java) || type.isAssignableFrom(java.lang.Byte::class.java) || type.isEnum) {
+            input.toString()
+        } else if (type == String::class.java) {
+            """"$input""""
```

Comment:
I believe backslashes and double quotes need to be escaped here.

Reply:
It's a multiline string, so I don't think any escaping is necessary. But maybe I'm not understanding what you're pointing out?

Reply:
The escapes need to make it out to the serialized request, otherwise once it becomes GraphQL it will look like this:
```
inputParam: "my"unescaped"string\"
```

Reply:
Added a test: https://github.com/Netflix/dgs-framework/pull/333/commits/a762cb7656376cc4be9250cdae5e6e281fe40e8d

I think that's what you're describing?

Reply:
Yeah, that's right - if that works then we're all good!  Thanks Paul.

Reply:
So actually, this test case does not show the solution, it shows the problem.  Those need to be double escaped so that the escapes are added for the query.  Try running your test case example through the DgsQueryExecutor and it will get confused because of stray quotes and backslashes in the middle of the String.